src/main/java/com/squareup/javapoet/WildcardTypeName.java

```diff
@@ -42,6 +43,7 @@ private WildcardTypeName(List<TypeName> upperBounds, List<TypeName> lowerBounds,
     super(annotations);
     this.upperBounds = Util.immutableList(upperBounds);
     this.lowerBounds = Util.immutableList(lowerBounds);
+    this.hashCode = upperBounds.hashCode() ^ lowerBounds.hashCode();
```

Comment:
Yes and no. The precalc of hash code in `ParameterizedTypeName` is neccessary - the precalcs in `TypeVariableName` and `WildcardTypeName` are here, to a) make it look and work similar and b) prevent future stack overflows.